db/migrate/20180917004159_create_transactions.rb

```diff
@@ -0,0 +1,22 @@
+class CreateTransactions < ActiveRecord::Migration[5.1]
+  def change
+    create_table :accounts do |t|
+      t.string :name
+      t.timestamps
+    end
+    create_table :transactions do |t|
+      t.integer :type
+      t.integer :status, default: 0
+      t.references :origin_account_id,
```

Comment:
this is my second or third time seeing this referenced i saw it in the associations above I have not seen this pattern before and I am not sure it wont lead to some problems.  Unless there is good reasoning or a working example where this works I suggest we use types or types with single table inheritance.